src/template.mligo

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: 2021 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+// Replace TEMPLATE_H with something unique corresponding to your
+// variant. For example, if your variant is called MyBaseDAOVariant you
+// can change TEMPLATE_H to MYBASEDAOVARIANT_H so this section reads
+//
+// #if !MYBASEDAOVARIANT_H
+// #define MYBASEDAOVARIANT_H
+
+#if !TEMPLATE_H
+#define TEMPLATE_H
+
+//
+
+#include "types.mligo"
+
```

Comment:
:thinking: how come `contract_extra` is not variant-specific as well?

AFAIK that field has a different content depending on the contract, but one that will be fixed throughout its life.

This would save a lot of unnecessary `pack`/`unpack` calls and be overall easier to use, I think.

IIRC this was sort of what we did back when there was a `lorentz` version of baseDAO too BTW.

Reply:
I changed the LIGO contract to have differnet storages, but I am not able to find a way to make this change on the Haskell side. Specifically I cannot find a way to deriving `StorageRPC` instance for different variants of storage, without completely duplicating the common fields.

I mean, how can we accomodate the Storage variants here?

```
deriveRPCWithStrategy "Storage" ligoLayout
```

Can you think of a solution for this?

Reply:
IIUC you now have a single `Storage` type with an argument (for `contract_extra`) and a concrete datatype for each variant, right?

If that's the case, have you tried doing something similar to what [`morley-ledgers` does](https://gitlab.com/morley-framework/morley-ledgers/-/blob/3a7d14cff8ea011e79ceec0bcdc1668bf987f7e9/code/morley-ledgers/src/Lorentz/Contracts/ManagedLedger/Types.hs#L110-L114)?

In this case you would have something like:
```haskell
data Storage contract_extra = -- <definition>

data TrivialExtra = -- <definition>
type instance AsRPC TrivialExtra = TrivialExtra

data RegistryExtra = -- <definition>
type instance AsRPC RegistryExtra = RegistryExtra

data TreasuryExtra = -- <definition>
type instance AsRPC TreasuryExtra = TreasuryExtra


deriveRPC "Storage"

type TrivialStorage = Storage TrivialExtra
type TrivialStorageRPC = StorageRPC TrivialExtra

type RegistryStorage = Storage RegistryExtra
type RegistryStorageRPC = StorageRPC RegistryExtra

type TreasuryStorage = Storage TreasuryExtra
type TreasuryStorageRPC = StorageRPC TreasuryExtra
```

Reply:
:thinking: that sounds like there might be a mistake in the definition of `deriveRPC` itself.

Can you try to reproduce the issue with a minimal example?

Reply:
It indeed looks like we bumped in a bug here.

I tried to look into it a bit and I couldn't quickly find the root cause, so I opened [`morley/#754`](https://gitlab.com/morley-framework/morley/-/issues/754) with an even smaller reproducible example.

However it might take a while for that to be solved and for an update to be usable here.
In the meanwhile I think we should remove the use of `customGeneric`/`deriveRPCWithStrategy` and `ligoLayout` and instead sort the fields manually and use `deriving stock Generic`/`deriveRPC`, undoing (at least for this type) the work done in #259

Reply:
Did it cause complex changes?

I would have thought that such an update would have mostly required to rearrange fields and their uses :thinking:

Reply:
Oh, it was not about using stock derivation of Generic instance, but about the bigger task of switching to polymorphic storage on Haskell side.

Reply:
Done now.